### ISSUE TITLE
TTY is broken for Docker Compose V2 in v3.5.2

### ISSUE THREAD

**Issue by olvlvl** [docker/compose-cli#1933]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior

I can use arrows to go up and down bash's history, move the cursor left and right, and use Ctrl-C to quit a command without killing the container.

### Actual behavior

![image](https://user-images.githubusercontent.com/155998/126032976-a7aba56d-1c5f-4569-8903-51230fa3e81c.png)

### Information

  - macOS Version: 11.4
  - Intel chip or Apple chip: Both.
  - Docker Desktop Version: 3.5.2

### Steps to reproduce the behavior

  1. Run your favorite container
  2. Try the arrows

I installed v3.5.1 back and it's working as expected.
I installed v3.5.2 back and disabled Docker Compose V2 and it's working as expected.

**Comment by tbenderpayx** [comment]
I had this issue when using `docker compose run ...`  with `compose: Docker Compose (Docker Inc., v2.0.0-beta.6)`.

I manually reverted to `v2.0.0-beta.5` and the issue remained.

I manually reverted to `v2.0.0-beta.4` and this version does not have the issue.

I suspect/assume that it was the changes in the release notes for:

* compose run and compose exec use separate streams for stdout and stderr
